### DISCUSSION TITLE
Problem with VirtualBox performance.

### DISCUSSION THREAD

**Discussion by CloisteredNeuron** [labwc/labwc#1874] [Q&A]
I've installed Labwc (packages.debian.org/trixie/labwc) on a Debian 6.7.12-1 VirtualBox VM.  The VM has 4GB of RAM, 20GB of storage, and 2 processors.  The install line:

sudo apt install libc6 libcairo2 libglib2.0-0t64 libinput10 libpango-1.0-0 libpangocairo-1.0-0 libpixman-1-0 libpng16-16t64 librsvg2-2 libwayland-server0 libwlroots12t64 libxcb1 libxkbcommon0 libxml2 xwayland alacritty kanshi mako-notifier swaybg swayidle waybar

Image with labwc running and "Enable 3D Acceleration" set in VirtualBox.
https://i.postimg.cc/Gmj6PjQ9/vm-labwc.gif

I start labwc with:

labwc -s alacritty

Labwc starts but is very sluggish in even responding to mouse movements.  Is there anything I can do to improve performance.  I recently configured Debian with IceWM/XORG and it was very responsive in a VM with 300MB of RAM and two processors so I'm thinking there is something special I might need to do with wayland.

Thanks for taking a look at this issue.

**Comment by johanmalm** [comment]
I don't know and haven't run VirtualBox for a long time.

I ran with `qemu` recently and there was no sluggishness (some simple notes at https://github.com/johanmalm/liatach) so guess it's some setting.

**Comment by CloisteredNeuron** [comment]
Good to hear it is not sluggish in qemu so virtualization in general is not a problem.  Thanks!

**Comment by mi4code** [comment]
Had the same responsiveness issue, `WLR_RENDERER=pixman labwc` fixed it.
